File: backend/Marketplace/Marketplace.Service/Services/PaymentService.cs
using AutoMapper;
using Marketplace.API;
using Marketplace.Core.DTOs.Payment;
using Marketplace.Core.Entities;
using Marketplace.Core.Exceptions;
using Marketplace.Core.Response;
using Microsoft.EntityFrameworkCore;

namespace Marketplace.Service.Services
{
    public class PaymentService
    {
        private readonly ApplicationDbContext _context;
        private readonly IMapper _mapper;

        public PaymentService(ApplicationDbContext context, IMapper mapper)
        {
            _context = context;
            _mapper = mapper;
        }

        public async Task<ServiceResult<IEnumerable<PaymentResponse>>> GetPaymentAsync()
        {
            var payments = await _context.Payments.ToListAsync();
            var mapped = _mapper.Map<IEnumerable<PaymentResponse>>(payments);
            return ServiceResult<IEnumerable<PaymentResponse>>.Success(mapped);
        }

        public async Task<ServiceResult<PaymentResponse>> GetPaymentByOrderId(long id)
        {
            var payments = await _context.Payments
                .Where(x => x.OrderId == id)
                .ToListAsync();

            if (!payments.Any())
                return ServiceResult<PaymentResponse>.Failure("Payment not found", 404);

            var mapped = _mapper.Map<PaymentResponse>(payments);
            return ServiceResult<PaymentResponse>.Success(mapped);
        }

        public async Task<ServiceResult<bool>> AddPayment(PaymentCreateRequest paymentRequest)
        {
            var order = await _context.Orders.FirstOrDefaultAsync(o => o.Id == paymentRequest.OrderId);
            if (order == null)
                return ServiceResult<bool>.Failure("Order not found", 404);

            var payment = _mapper.Map<Payment>(paymentRequest);

            await _context.Payments.AddAsync(payment);
            await _context.SaveChangesAsync();

            return ServiceResult<bool>.Success(true, 201);
        }

        public async Task<ServiceResult<bool>> DeletePayment(long id)
        {
            var payment = await _context.Payments.FirstOrDefaultAsync(x => x.Id == id);
            if (payment == null)
                return ServiceResult<bool>.Failure("Payment not found", 404);

            _context.Payments.Remove(payment);
            await _context.SaveChangesAsync();

            return ServiceResult<bool>.Success(true, 200);
        }
    }
}

File: frontend/client-app/src/services/api/paymentService.ts
import axios from 'axios';
import { API_BASE_URL } from '@/constants/api';
import type { ApiResponse } from '@/types/ApiResponse';
import type { PaymentResponse } from '@/types/payment/PaymentResponse';
import type { PaymentCreateRequest } from '@/types/payment/PaymentCreateRequest';
import { createDefaultErrorResponse } from '@/utils/createDefaultErrorResponse';

const paymentService = {
  async getPayments(): Promise<ApiResponse<PaymentResponse[]>> {
    try {
      const response = await axios.get<ApiResponse<PaymentResponse[]>>(`${API_BASE_URL}/v1/payments`);
      return response.data;
    } catch (error: any) {
      return error.response?.data || createDefaultErrorResponse<PaymentResponse[]>();
    }
  },

  async getPaymentByOrderId(orderId: number): Promise<ApiResponse<PaymentResponse>> {
    try {
      const response = await axios.get<ApiResponse<PaymentResponse>>(`${API_BASE_URL}/v1/payments/order/${orderId}`);
      return response.data;
    } catch (error: any) {
      return error.response?.data || createDefaultErrorResponse<PaymentResponse>();
    }
  },

  async createPayment(request: PaymentCreateRequest): Promise<ApiResponse<null>> {
    try {
      const response = await axios.post<ApiResponse<null>>(`${API_BASE_URL}/v1/payments`, request);
      return response.data;
    } catch (error: any) {
      return error.response?.data || createDefaultErrorResponse<null>();
    }
  },

  async deletePayment(id: number): Promise<ApiResponse<null>> {
    try {
      const response = await axios.delete<ApiResponse<null>>(`${API_BASE_URL}/v1/payments/${id}`);
      return response.data;
    } catch (error: any) {
      return error.response?.data || createDefaultErrorResponse<null>();
    }
  },
};

export default paymentService;
